Charandeep- Progress Bar Page

charandeep-k opened the pull request:
Home Page:

Implemented pagination for student list to improve navigation and scalability.

Enhanced data fetching with page/limit parameters for optimized performance.

ProgressBar Page:

Added a dialogue box component for Topics.

Integrated  filters inside the dialogue Box.

Attached Images for Reference

<img width="959" height="495" alt="dailogue box" src="https://github.com/user-attachments/assets/676c8ef1-a45b-4b0d-84ac-cb07aa85df25" />


<img width="959" height="497" alt="Home Page Pagination" src="https://github.com/user-attachments/assets/4a132f04-260b-4ae8-88a1-9a297732c3ab" />

Commit:
Added Components into Progress Page

Commit:
added dialog box, enhanced filters & improved ProgressBar UX,
implemented pagination in Home page for student list

Commit:
changes: added scrollbar to student list in the home page

sahilsablani-17 commented on Client/app/src/components/CourseCardPb.tsx:
```diff
@@ -38,7 +44,17 @@ const CourseCard: React.FC<CourseCardProps> = ({ courses }) => {
               >
-                <td>{course.course_name}</td>
+                <td
+                  style={{ cursor: "pointer", color: "#1976d2" }}
```
Make style in .css file

charandeep-k replied:
I have pushed styles into CSS

sahilsablani-17 commented on Client/app/src/pages/ProgressBar.tsx:
```diff
@@ -1,19 +1,72 @@
+            ) : courses.length === 0 ? (
+          <p>No courses found for {selectedStudent.name}</p>
+) : (
+  <CourseCard courses={courses} />
```
Make formatting little better

charandeep-k replied:
It is already been updated

Commit:
changes: pushed styles into css, sorted sidebar .

Commit:
changes : added clear filter button on progress bar page

SudhanvaCodes commented:
<img width="1333" height="916" alt="image" src="https://github.com/user-attachments/assets/772e53cd-cd5e-4ae9-ac68-609cc8a755ca" /><img width="1339" height="952" alt="image" src="https://github.com/user-attachments/assets/8b056106-b7b0-40e8-b9e1-4276911a4fc4" />
I see the following screens when I refresh http://localhost:3000/progress. Please fix these by adding a circular loader instead of showing no courses found, also check why the other text is coming on load

SudhanvaCodes commented:
<img width="1337" height="905" alt="image" src="https://github.com/user-attachments/assets/57ad891c-6b23-4af8-81fa-4981c48a459e" />
Add a circular loader each time I click on different user until data is fetched, else it feels like theres some lag in UI

SudhanvaCodes commented:
Center align the loader. Also check why the modal size is small during loading
<img width="638" height="306" alt="image" src="https://github.com/user-attachments/assets/2d9ee2ab-0ebc-4fa5-bd84-710e07e172fc" />

SudhanvaCodes commented:
The modal size is inconsistent when data is present versus when it’s not. Please make the modal size uniform
<img width="640" height="344" alt="image" src="https://github.com/user-attachments/assets/976917c7-50e9-419a-aebf-805259da67ad" />

SudhanvaCodes commented:
<img width="270" height="90" alt="image" src="https://github.com/user-attachments/assets/7173dd13-d4fa-4129-88ab-559764daeb90" />
Add reset icon too along with this clear filters text

SudhanvaCodes commented:
<img width="1341" height="912" alt="image" src="https://github.com/user-attachments/assets/8c944005-60af-4c22-9943-7f63ae12c35c" />
Center align this to page and increase font size

SudhanvaCodes commented:
<img width="1543" height="339" alt="image" src="https://github.com/user-attachments/assets/553aa117-173f-4c96-9135-059c93aa0438" />
Let the cursor be pointer only on course names not on other attributes

SudhanvaCodes commented on Client/app/src/components/CourseTopicsDialog.tsx:
```diff
@@ -0,0 +1,140 @@
+  courseName,
+}) => {
+  const [topics, setTopics] = useState<Topic[]>([]);
+  const [filteredTopics, setFilteredTopics] = useState<Topic[]>([]);
```
filteredTopics var is no where used

SudhanvaCodes commented on Client/app/src/components/CourseTopicsDialog.tsx:
```diff
@@ -0,0 +1,140 @@
+    if (!status || status === "all") return true;
+    if (status === "Pending") return topic.status  !== "Completed";
+    return topic.status === status;
+  });
```
memoize visibleTopics

Commit:
Added Enrollment Chart and search bar

Commit:
changes:
Added dialog box for course topics with fixed width and memoized topic filtering.
Updated Clear Filters button to include a reset icon for better UX.
Fixed loader duplication issue on the Progress page.
Aligned “No courses found” message vertically and horizontally in the courses container.
Refactored filter logic and memoized filteredCourses for performance.
Cleaned up inline styles in CourseCard and moved them to CSS.
Ensured cursor pointer applies only to course names in the table.
Adjusted dialog box size to prevent shrinking or growing unexpectedly.

charandeep-k commented:
**Enhance Progress Page: Dialog Box, Filters, Loader & Reset Icon**

Added dialog box for course topics with fixed width and memoized topic filtering.

Updated Clear Filters button to include a reset icon for better UX.

Fixed loader duplication issue on the Progress page.

Aligned “No courses found” message vertically and horizontally in the courses container.

Refactored filter logic and memoized filteredCourses for performance.

Cleaned up inline styles in CourseCard and moved them to CSS.

Ensured cursor pointer applies only to course names in the table.

Adjusted dialog box size to prevent shrinking or growing unexpectedly.

Commit:
changes: code formatting for all files , Added progress bar component and enhanced it

Commit:
removed unwanted comments

Commit:
enhanced layouts , removed unwanted imports
